plugins/modules/azure_rm_notificationhub.py

```diff
@@ -0,0 +1,405 @@
+            log_path=dict(type='str'),
+            log_mode=dict(type='str'),
+        )
+
+        self.resource_group = None
+        self.namespace_name = None
+        self.name = None
+        self.sku = None
+        self.location = None
+        self.authorizations = None
+        self.tags = None
+        self.state = None
+        self.log_path = None
+        self.log_mode = None
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        super(AzureNotificationHub, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                   supports_check_mode=True,
+                                                   supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # retrieve resource group to make sure it exists
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            # Set default location
+            self.location = resource_group.location
+
+        results = dict()
+        changed = False
+
+        try:
+            self.log(
+                'Fetching Notification Hub Namespace {0}'.format(self.name))
+            namespace = self.notification_hub_client.namespaces.get(
+                self.resource_group, self.namespace_name)
+
+            results = namespace_to_dict(namespace)
+            if self.name:
+                self.log('Fetching Notification Hub {0}'.format(self.name))
+                notification_hub = self.notification_hub_client.notification_hubs.get(
+                    self.resource_group, self.namespace_name, self.name)
+                notification_hub_results = notification_hub_to_dict(
+                    notification_hub)
+            # don't change anything if creating an existing namespace, but change if deleting it
+            if self.state == 'present':
+                changed = False
+
+                update_tags, results['tags'] = self.update_tags(
+                    results['tags'])
+
+                if update_tags:
+                    changed = True
+                elif self.namespace_name and not self.name:
+                    if self.sku != results['sku'].lower():
+                        changed = True
+
+            elif self.state == 'absent':
+                changed = True
+
+        except CloudError:
+            # the notification hub does not exist so create it
+            if self.state == 'present':
+                changed = True
+            else:
+                # you can't delete what is not there
+                changed = False
+
+        self.results['changed'] = changed
+        if self.name and not changed:
+            self.results['state'] = notification_hub_results
+        else:
+            self.results['state'] = results
```

Comment:
Please delete line 235 ~ 238, Adjust line 239. Thank you very much!
```suggestion
        self.results['state'] = results
```